sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1714,29 +1714,39 @@ object GpuOverrides {
           (childExprs.head.dataType, childExprs(1).dataType) match {
             case (l: DecimalType, r: DecimalType) =>
               val outputType = GpuDivideUtil.decimalDataType(l, r)
-              // We will never hit a case where outputType.precision < outputType.scale + r.scale.
-              // So there is no need to protect against that.
-              // The only two cases in which there is a possibility of the intermediary scale
-              // exceeding the intermediary precision is when l.precision < l.scale or l
-              // .precision < 0, both of which aren't possible.
-              // Proof:
-              // case 1:
-              // outputType.precision = p1 - s1 + s2 + s1 + p2 + 1 + 1
-              // outputType.scale = p1 + s2 + p2 + 1 + 1
-              // To find out if outputType.precision < outputType.scale simplifies to p1 < s1,
-              // which is never possible
-              //
-              // case 2:
-              // outputType.precision = p1 - s1 + s2 + 6 + 1
-              // outputType.scale = 6 + 1
-              // To find out if outputType.precision < outputType.scale simplifies to p1 < 0
-              // which is never possible
+              // Case 1: OutputType.precision doesn't get truncated
+              //   We will never hit a case where outputType.precision < outputType.scale + r.scale.
```

Comment:
I think it would be good (and fairly easy) to add a unit test for this. I had a go and found some values that do hit this case but it is possible that I am using values that couldn't happen in real life. I am not sure.

Here is one example:

```
l = DecimalType(3,0)
r = DecimalType(21,17)
outputType = DecimalType(38,23)
38 < 23 + 17
```

Reply:
Perhaps I am hitting case 2 here where precision is getting truncated?

Reply:
It does look like I am hitting case 2 here so I think the logic in the comments is sound, but a unit test would make me more confident and would protect against future regressions if any of this code gets updated.

Reply:
+1 for the unit test to verify that this is working.  The logic looks good to me. I am a little concerned about the coupling between this code and the code in `GpuDivideUtil`, but that is minor for now.

Reply:
Yes, I wrote a unit test locally to verify this. I don't know why I didn't check it in. Let me do that

Reply:
> l = DecimalType(3,0)
r = DecimalType(21,17)
outputType = DecimalType(38,23)
38 < 23 + 17
So, like you said this won't happen in reality as by the time we get the call both `l` and `r` Spark will be the same precision and scale, I think it will upcast the Decimal(3,0) to Decimal(21,17).